src/d/actor/d_a_obj_ohatch.cpp

```diff
@@ -4,72 +4,324 @@
+    fopAcM_SetupActor(this, daObjOhatch_c);
+    cPhs_State ret = dComIfG_resLoad(&mPhase, l_arcname);
+    if (ret == cPhs_COMPLEATE_e) {
+        if (fopAcM_entrySolidHeap(this, solidHeapCB, 0xB90)) {
+            fopAcM_SetMtx(this, mModel->getBaseTRMtx());
+            m29C = param_get_swSave();
+            if (m29C != 0xff && fopAcM_isSwitch(this, m29C) == TRUE) {
+                m2D8 = -0x4000;
+                mActionFunc = &daObjOhatch_c::open_wait_act_proc;
+                if (dComIfG_Bgsp()->Regist(mBgW2, this)) {
+                    ret = cPhs_ERROR_e;
+                }
+            } else {
+                mActionFunc = &daObjOhatch_c::close_wait_act_proc;
+                if (dComIfG_Bgsp()->Regist(mBgW1, this)) {
+                    ret = cPhs_ERROR_e;
+                }
+            }
+            init_mtx();
+        } else {
+            ret = cPhs_ERROR_e;
+        }
+    }
+
+#if VERSION == VERSION_DEMO
+    if (l_HIO.m04 < 0) {
+        l_HIO.m04 = mDoHIO_createChild("フィギュア屋の入口", &l_HIO);
+    }
+#endif
+    return ret;
 }
 
 /* 00000548-00000628       .text _delete__13daObjOhatch_cFv */
 bool daObjOhatch_c::_delete() {
-    /* Nonmatching */
+    dComIfG_resDelete(&mPhase, l_arcname);
+#if VERSION > VERSION_DEMO
+    if (heap != NULL)
+#endif
+    {
+        if (mBgW1 != NULL) {
+            if (mBgW1->ChkUsed()) {
+                dComIfG_Bgsp()->Release(mBgW1);
+            }
+#if VERSION > VERSION_DEMO
+            mBgW1 = NULL;
+#endif
+        }
+
+        if (mBgW2 != NULL) {
+            if (mBgW2->ChkUsed()) {
+                dComIfG_Bgsp()->Release(mBgW2);
+            }
+#if VERSION > VERSION_DEMO
+            mBgW2 = NULL;
+#endif
+        }
+    }
+
+#if VERSION == VERSION_DEMO
+    if (l_HIO.m04 >= 0) {
+        mDoHIO_deleteChild(l_HIO.m04);
+        l_HIO.m04 = -1;
+    }
+#endif
+    return true;
 }
 
 /* 00000628-00000810       .text close_wait_act_proc__13daObjOhatch_cFv */
 void daObjOhatch_c::close_wait_act_proc() {
-    /* Nonmatching */
+    if (m29C != 0xff && fopAcM_isSwitch(this, m29C) == TRUE) {
+        if (dComIfGp_evmng_existence("FIGURE_HATCH_OPEN")) {
+            s32 staffIdx = dComIfGp_evmng_getMyStaffId("Ohatch");
+            if (staffIdx != -1) {
+                if (strcmp(dComIfGp_getPEvtManager()->getMyNowCutName(staffIdx), "OPEN") == 0) {
+                    fopAcM_seStartCurrent(this, JA_SE_OBJ_FIG_HATCH_OPEN, 0);
+                    if (mBgW1 != NULL && mBgW1->ChkUsed()) {
+                        dComIfG_Bgsp()->Release(mBgW1);
+                    }
+                    if (mBgW2 != NULL && !mBgW2->ChkUsed()) {
+                        dComIfG_Bgsp()->Regist(mBgW2, this);
+                    }
+
+                    dComIfGp_getVibration().StartShock(3, 1, cXyz(0.0f, 1.0f, 0.0f));
+                    mActionFunc = &daObjOhatch_c::tremor_act_proc;
+                }
+            }
+        }
+    }
+#if VERSION == VERSION_DEMO
+    else if (l_HIO.m38 == 1)
```

Comment:
Does this still match if you remove the ifdef? I'd assume so since checking if a const 0 equals 1 should optimize out the entire block in retail.
```suggestion
    else if (HIO(m38) == 1)
```